language-server/src/server.js

```diff
@@ -148,43 +179,70 @@ documents.onDidChangeContent(async ({ document }) => {
   const instance = JsoncInstance.fromTextDocument(document);
-  const $schema = instance.get("#/$schema");
-  const contextDialectUri = $schema.value();
-  const schemaResources = decomposeSchemaDocument(instance, contextDialectUri);
-  for (const { dialectUri, schemaInstance } of schemaResources) {
-    if (!hasDialect(dialectUri)) {
-      const $schema = schemaInstance.get("#/$schema");
-      if ($schema.typeOf() === "string") {
-        diagnostics.push(buildDiagnostic($schema, "Unknown dialect"));
-      } else {
-        diagnostics.push(buildDiagnostic(schemaInstance, "No dialect"));
+  try {
+    const $schema = instance.get("#/$schema");
+   // Determine the context dialect URI from the document
+    contextDialectUri = $schema?.value();
+    dialectUri = contextDialectUri;
+
+    const schemaResources = decomposeSchemaDocument(instance, contextDialectUri);
+    for (const { dialectUri, schemaInstance } of schemaResources) {
+      if (!hasDialect(dialectUri)) {
+        const $schema = schemaInstance.get("#/$schema");
+        if ($schema) {
+          diagnostics.push(buildDiagnostic($schema, "Unknown dialect"));
+        } else {
+          diagnostics.push(buildDiagnostic(schemaInstance, "No dialect"));
+        }
+
+        continue;
       }
 
-      continue;
-    }
-
-    const [output, annotations] = await validate(dialectUri, schemaInstance);
+      const [output, annotations] = await validate(dialectUri, schemaInstance);
 
-    if (!output.valid) {
-      for await (const [instance, message] of invalidNodes(output)) {
-        diagnostics.push(buildDiagnostic(instance, message));
+      if (!output.valid) {
+        for await (const [instance, message] of invalidNodes(output)) {
+          diagnostics.push(buildDiagnostic(instance, message, settings.schemaValidationSeverity === "warning" ? DiagnosticSeverity.Warning : DiagnosticSeverity.Error));
+        }
       }
-    }
 
-    const deprecations = annotations.annotatedWith("deprecated");
-    for (const deprecated of deprecations) {
-      if (deprecated.annotation("deprecated").some((deprecated) => deprecated)) {
-        const message = deprecated.annotation("x-deprecationMessage").join("\n") || "deprecated";
-        diagnostics.push(buildDiagnostic(deprecated.parent(), message, DiagnosticSeverity.Warning, [DiagnosticTag.Deprecated]));
+      const deprecations = annotations.annotatedWith("deprecated");
+      for (const deprecated of deprecations) {
+        if (deprecated.annotation("deprecated").some((deprecated) => deprecated)) {
+          const message = deprecated.annotation("x-deprecationMessage").join("\n") || "deprecated";
+          diagnostics.push(buildDiagnostic(deprecated.parent(), message, DiagnosticSeverity.Warning, [DiagnosticTag.Deprecated]));
+        }
       }
     }
+  } catch (error) {
+    const defaultDialect = settings.defaultDialect;
+    dialectUri = defaultDialect;
   }
 
+
+  connection.console.log(`Schema used: ${dialectUri}`);
```

Comment:
While I'm sure this was useful in debugging, it should be removed before this is merged.

```suggestion
```